castle/cms/static/patterns/sticky.js

```diff
@@ -12,13 +13,31 @@ define([
     parser: 'mockup',
     defaults: {},
     init: function() {
+      var self = this;
       window.addEventListener("load", function(event) {
-        $(".sticky-footer").show();
-        $(".sticky-footer")
-          .css({ bottom: "-100px" })
-          .animate({ bottom: "0px" }, "slow");
+        self.cookie_data = $.cookie("sticky-footer");
+
+        // Sets the cookie to indicate the footer was closed and won't re-appear.
+        $(".close").click(function() {
+          self.cookie_data = self.getCookie("closed-footer")
+        });
+
+        if (self.cookie_data !== "closed-footer") {
+          $(".sticky-footer").show();
+            // If cookie hasn't been set yet, do the rising animation.
+            if (typeof(self.cookie_data) === "undefined") {
+              $(".sticky-footer")
+                .css({ bottom: "-100px" })
+                .animate({ bottom: "0px" }, "slow");
+              self.cookie_data = self.getCookie("no-animation");
+          }
+      }
       });
-    }
+    },
+
+    getCookie: function(args) {
```

Comment:
I think this can be renamed to setCookie since it is setting the cookie value (it must return the new value anyway such that the logic works).  Line 18 $.cookie("sticky-footer") is where it actually gets any existing value for the cookie.   I think just renaming the function and calls to it will be good.